src/sys/mod.rs

```diff
@@ -34,3 +34,27 @@ pub use self::imp::*;
 
 #[path = "linux/mod.rs"]
 mod imp;
+
+
+#[doc(hidden)]
+pub trait AsInner<Inner: ?Sized> {
```

Comment:
`AsInner` and friends are usually part of `std::sys_common` in the normal standard library. I moved them into `sys/unix` instead, though they could be moved back if we do end up introducing the `sys_common` module in this library.